### DISCUSSION TITLE
Why two calendar tables ?

### DISCUSSION THREAD

**Discussion by bdarbonneau** [sql-bi/Bravo#638] [Q&A]
When Bravo creates a calendar table, it creates a hidden calculated table, then another one that refers to the first one. The model eventually contains two identical tables, one of them not used but which still consumes memory. Why not create a single one ? 
I am curious to understand the reason behind this design choice.

**Comment by albertospelta** [comment]
Hi @bdarbonneau, the reason for this choice was to try to avoid the PowerBI Desktop behavior which automatically expands the expression editor whenever you select a column from a calculated table. This behavior is quite annoying as the expression editor completely hides the report view in case the expression contains many lines of DAX code. For this reason we decided to hide the main table that calculates dates and use a second table containing a single line of DAX code.

Thanks for point this out! I'm going to create an issue for this (see https://github.com/sql-bi/Bravo/issues/639) as this may be an enhancement we may evaluate for a feature version in case PBIDesktop changes this behavior.

![image](https://user-images.githubusercontent.com/1957788/231377580-6727e38c-707a-4930-a4f6-151c64a9f50d.png)

**Comment by bdarbonneau** [comment]
Hi Alberto,
This auto-expansion is indeed a real pain. I hope too that Microsoft are going to do something about it.
Thks for the quick response.

Regards,
Bertrand
